### DISCUSSION TITLE
Should we add namespaces?

### DISCUSSION THREAD

**Discussion by max-ishere** [elkowar/eww#987] [Ideas]
This sounds a little out there for Yuck and Eww, but I think having a namespace would be nice. Essentially you would have a couple of options. Either a `(import "bar.yuck" as bar)` or (this would be way better) `(namespace bar)`. Then when you want to use a widget/window from a namespace you simply use `bar::Window`.

You can already name windows and widgets using `::`, but this will just make the process a little easier.

The issue with using `(import` is that it doesn't actually achieve anything since no renaming happens if at least one file in the include tree doesnt use the new `(import` syntax

**Comment by mantacid** [comment]
I don't think something like this would be bad. If the namespace had some effect on variable scoping then  it would be really good for storing local values, seeing as we can't really scope a variable to only be defined within a window.

Also, how did you find out about widget renaming? I can't remember the documentation mentioning that at all.

**Reply by max-ishere** [reply]
There's no renaming but I tried to use `::` and `.` In the name and it just worked.
